### PR TITLE
Use channels instead of mutex

### PR DESCRIPTION
**Pull request by ishitatsuyuki**
ALSA and WASAPI backends now doesn't use a Mutex that is locked every time data is processed. These two backends were straightforward to modify, other backends were not modified.

Changes are self descriptive.

### CODE REVIEW

File: src/wasapi/stream.rs [hunk 1]
```diff
@@ -405,14 +404,7 @@ impl EventLoop {
 
     #[inline]
     pub fn destroy_stream(&self, stream_id: StreamId) {
-        unsafe {
-            self.commands
-                .lock()
-                .unwrap()
-                .push(Command::DestroyStream(stream_id));
-            let result = synchapi::SetEvent(self.pending_scheduled_event);
```
Comment (mitchmindtree) [inline src/wasapi/stream.rs:413]: Just double checking, do we not need this anymore? I'm not familiar enough to remember the purpose of this line - is it to wake up the processing if stopped perhaps?

File: src/wasapi/stream.rs [hunk 2]
```diff
@@ -618,17 +611,19 @@ impl EventLoop {
 
     #[inline]
     pub fn play_stream(&self, stream: StreamId) {
-        unsafe {
-            self.commands.lock().unwrap().push(Command::PlayStream(stream));
-            let result = synchapi::SetEvent(self.pending_scheduled_event);
```
Comment (mitchmindtree) [inline src/wasapi/stream.rs:623]: Same with this

File: src/wasapi/stream.rs (hunk 1 above)
Comment (ishitatsuyuki) [reply src/wasapi/stream.rs:413]: I did a silent refactor to remove the duplicates - the logic is moved to a new `push_command` function.

Comment (mitchmindtree) [reply src/wasapi/stream.rs:413]: Ahhh yes I see now, my bad, I missed it on initial review!

**Pull request merged**